CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc

```diff
@@ -1668,20 +1778,280 @@ void EcalTPGParamBuilder::analyze(const edm::Event& evt, const edm::EventSetup&
+      if (weights_odd[igrp].size() == 5) {
+        if (writeToFiles_) {
+          (*out_file_) << std::endl;
+          (*out_file_) << "WEIGHT_ODD " << dec << igrp << endl;
+          for (unsigned int sample = 0; sample < 5; sample++)
+            (*out_file_) << "0x" << hex << weights_odd[igrp][sample] << " ";
+          (*out_file_) << std::endl;
+          (*out_file_) << std::endl;
+        }
+        if (writeToDB_) {
+          ss << "going to write the weights for group:" << igrp << "\n";
+          FEConfigOddWeightGroupDat gut;
+          gut.setWeightGroupId(igrp);
+          //PP WARNING: weights order is reverted when stored in the DB
+          gut.setWeight0(weights_odd[igrp][4]);
+          gut.setWeight1(weights_odd[igrp][3] +
+                         0x80);  //0x80 to identify the max of the pulse in the FENIX (doesn't exist in emulator)
+          gut.setWeight2(weights_odd[igrp][2]);
+          gut.setWeight3(weights_odd[igrp][1]);
+          gut.setWeight4(weights_odd[igrp][0]);
+          EcalLogicID ecid = EcalLogicID("DUMMY", igrp, igrp);  //1 dummy ID per group
+          // Fill the dataset
+          dataset_odd[ecid] = gut;
+        }
+      }
+    }
+    // TP MODE configuration
+    if (writeToFiles_) {
+      (*out_file_) << std::endl;
+      (*out_file_) << "TP_MODE " << endl;
+      (*out_file_) << TPmode_EnableEBOddFilter_ << endl;
+      (*out_file_) << TPmode_EnableEEOddFilter_ << endl;
+      (*out_file_) << TPmode_EnableEBOddPeakFinder_ << endl;
+      (*out_file_) << TPmode_EnableEEOddPeakFinder_ << endl;
+      (*out_file_) << TPmode_DisableEBEvenPeakFinder_ << endl;
+      (*out_file_) << TPmode_DisableEEEvenPeakFinder_ << endl;
+      (*out_file_) << TPmode_FenixEBStripOutput_ << endl;
+      (*out_file_) << TPmode_FenixEEStripOutput_ << endl;
+      (*out_file_) << TPmode_FenixEBStripInfobit2_ << endl;
+      (*out_file_) << TPmode_FenixEEStripInfobit2_ << endl;
+      (*out_file_) << TPmode_FenixEBTcpOutput_ << endl;
+      (*out_file_) << TPmode_FenixEBTcpInfobit1_ << endl;
+      (*out_file_) << TPmode_FenixEETcpOutput_ << endl;
+      (*out_file_) << TPmode_FenixEETcpInfobit1_ << endl;
+      (*out_file_) << 0 << endl;  // FenixPar15-18 are placeholder parameters for future use
+      (*out_file_) << 0 << endl;
+      (*out_file_) << 0 << endl;
+      (*out_file_) << 0 << endl;
+      (*out_file_) << std::endl;
+    }
+
+    if (writeToDB_) {
+      // now we store in the DB the correspondence btw channels and groups
+      map<EcalLogicID, FEConfigOddWeightDat> dataset_odd_idmap;
+
+      // EB loop
+      for (int ich = 0; ich < (int)my_StripEcalLogicId.size(); ich++) {
+        FEConfigOddWeightDat wut;
+        int igroup = weights_map_odd[stripMapEB[ich]];  // this group is for EB
+        wut.setWeightGroupId(igroup);
+        dataset_odd_idmap[my_StripEcalLogicId[ich]] = wut;
+      }
+      // EE loop
+      for (int ich = 0; ich < (int)my_StripEcalLogicId1_EE.size(); ich++) {
```

Comment:
range based loop can be preferred here, also to avoid these (useless) type castings